packages/main/src/plugin/cli-tool-registry.ts

```diff
@@ -82,17 +89,52 @@ export class CliToolRegistry {
     this.apiSender.send('cli-tool-remove', cliTool.id);
   }
 
-  getCliToolInfos(): CliToolInfo[] {
-    return Array.from(this.cliTools.values()).map(cliTool => {
-      return {
+  async getCliToolInfos(): Promise<CliToolInfo[]> {
+    const cliToolInfos: CliToolInfo[] = [];
+    for (const cliTool of this.cliTools.values()) {
+      const cliVersion = await cliTool.getVersion();
+      cliToolInfos.push({
         id: cliTool.id,
         name: cliTool.name,
         displayName: cliTool.displayName,
         description: cliTool.markdownDescription,
         state: cliTool.state,
         images: cliTool.images,
         extensionInfo: cliTool.extensionInfo,
-      };
-    });
+        version: cliVersion,
+      });
+    }
+    return cliToolInfos;
   }
 }
+
+// This function will take in the exec, name and storage path of the binary.
+// it will also take in the "command" that achieves getting the version of the binary as
+// each tool has a different way of getting it (for example, -v, --version, version, --help, etc.)
+// NOTE: We only care what's in the ~/.local/share/containers/podman-desktop/extensions-storage/ folder
+// not the binary that is installed on the system (since we have the ability to install system-wide)
+async function getCliVersion(exec: Exec, name: string, storagePath: string, helpCommand: string): Promise<string> {
+  // Make sure we resolve if it is a Windows binary or not as well.
+  const destFile = path.resolve(storagePath, 'bin', isWindows() ? name + '.exe' : name);
```

Comment:
It's because the ~/.local directory is universal between OS' and we provide the option to system-install. The .local directory 

Checking the version via $PATH may give different results depending on where the binary is located (could be located in three different areas... for example, the `kompose` binary may be located in the $GOBIN directory, or /usr/local/ or maybe even in brew).

So we would check the ~/.local folder first and then we can provide a way to install system-wide.

Reply:
I agree with @lstocchi that at this point storagePath should not be involved in detection, because if we provide binaries inside extension then we now the location and version. We also can say for sure if version installed system wide is the same we have bundled with extension